bash: update to 5.0

dgilman opened the pull request:
#### Description
Testing: I started a new shell, and did `while true;  do echo "hello world"; sleep 1; done`

###### Type(s)
- [x] enhancement

###### Tested on
macOS 10.13.6 17G4015
Xcode 10.1 10B61

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
<!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

pmetzger commented:
Hey, Travis failed on this, any idea why?

dgilman commented:
The bashdb is a debugger that uses header files from bash to parse bash's internal data structures. It must be revbumped when bash is updated. The latest bashdb doesn't support bash 5 yet, so its build failed. This PR shouldn't be closed until bashdb updates otherwise macports will break users of bashdb.

pmetzger commented:
I added the WIP tag to note that we're waiting for the `bashdb` update. We'll need @raimue to comment in any case.

raimue commented on shells/bash/Portfile:
```diff
@@ -23,89 +23,16 @@ master_sites        gnu
 
-checksums-append    bash44-001 \
-                    rmd160  a4915a389d04165402193ab681ed975b3e8a29d2 \
-                    sha256  3e28d91531752df9a8cb167ad07cc542abaf944de9353fe8c6a535c9f1f17f0f \
```
I would like to keep one checksum entry in a comment to make it easier to restore this functionality when new patches for bash 5.0 are released.

dgilman commented:
I tested building bash44 and bash (now 5.0) and they built fine. I did the subport in a very imperative style with lots of conditionals, if you think it could/should be done in a more object-oriented style I could do that if you provide some feedback on how to do it.

I didn't test the bashdb upgrade path, hopefully this doesn't cause too much pain with the subport conflicts for users of bashdb who would have already installed bash?

pmetzger commented:
@raimue This is all your call.

dgilman commented:
I've updated the Portfile for the two new patches that came out over the weekend. There was also some inconsistency with tabs and spaces, I switched everything over to spaces.

pmetzger commented:
@raimue ?

raimue commented:
It's probably not the best that bash and bash44 conflict, but let's go with it in the hope bashdb gets updated and we can get rid of bash44 later.

Thanks for all your work on this. Sorry I am so slow to review these at the moment. I appreciate the contribution!

raimue commented:
They could be made to install side-by-side by separating the files and installing the binary as something like `bash44`. But then bashdb would also require patches to find the different bash version... This would allow to use bashdb and bash 5.0 at the same time, but maybe this is also not that important. As I said, I hope bashdb gets updated and just for the interim time it will not be worth the required effort.
